lib/internal/modules/esm/loader.js

```diff
@@ -45,6 +46,16 @@ class Loader {
     // an object with the same keys as `exports`, whose values are get/set
     // functions for the actual exported values.
     this._dynamicInstantiate = undefined;
+
+    // Set up context passed to hooks
+    const k = Object.freeze(Object.create(null));
+    vmModuleLinkHookMap.set(k, this);
+
+    this.hookContext = Object.assign(Object.create(null), {
```

Comment:
I'd prefer a different name since this has relations to `vm` and `Context` has a meaning of a Realm in the `vm` terminology.

This also seems to have some conflict with https://github.com/nodejs/node/pull/18914 that I need to think on.